### DISCUSSION TITLE
定制导入到 flomo 的模板

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2275] [服务]
> 简悦 2.2.0 支持导入到 flomo 细节 [请看这里](https://github.com/Kenshin/simpread/discussions/2150)

> 此方式的由于是简悦官方给出的结构，所以无法修改，如果你需要 **定制化导入到 flomo 的结构**，可以看看本篇教程。

## 关联阅读

> 此功能使用了 **[Webhook](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=webhook)** 与 **[Markdown 定制化](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown)**

## Webhook 配置

> 使用了简悦的一个第三方 API 服务器，将下列代码加入到 Webhook 中，注意下面的代码需要压缩为一行，可以利用 [这个在线服务](https://tool.lu/json/)

> 在使用此 API 时，务必确保 **网络畅通**。

```
{
    "name": "Webhook2flomo",
    "url": "https://api-wrap.simpread.pro/api/service/flomo",
    "type": "POST",
    "fmt": "md",
    "body": {
        "endpoint": "https://flomoapp.com/iwh/*****/*****/",
        "template": "#收集\nnote [[note]]\n[[content]]\n------《[[title]]》[[url]] [[tags]]",
        "params": {
            "url": "{{url}}",
            "title": "{{title}}",
            "desc": "{{desc}}",
            "note": "{{note}}",
            "tags": "{{tags}}",
            "content": "{{content}}"
        }
    }
}
```

### 关键说明

- `name` → 可随便更改，仅支持 `英文 + 数字` 的方式
- `endpoint` 的值应为你在 [flomo API](https://flomoapp.com/mine?source=incoming_webhook) 中的 URL
- `template` 是你自己设置的模板，请留意其中 `[[xxx]]` 的结构，其中的 xxx 则为 `params` 对应的值，如：`[[url]]` → `{{url}}` 
- `params` 对应的字段，可根据自身调整，如不需要就去掉它，关于这些值具体的含义 [请看这里](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown)
- 除以之外的结构 **不可改变**

> 上述内容压缩后，加入到 选项页 → 服务 → Webhook，如下图所示：

![image](https://user-images.githubusercontent.com/81074/119977647-49b8ce00-bfeb-11eb-90bc-e1168d446fd3.png)

## 定制化 Markdown 配置

> 下面的代码格式化细节 [请看这里](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown-1)

> `<!--[[flomo]]-->` 必须存在

```
<!--[[flomo]]-->

{{desc}}

{{note}}

{{tags}}

{{#each}}

{{an_html}}

{{an_note}}

{{an_tags}}

{{/each}}
```

> 将上面的代码粘贴到下图：

![image](https://user-images.githubusercontent.com/81074/120091709-50a52500-c140-11eb-8e0d-71b9b95a9871.png)

> 同时也可以配置 **单条标注** 时的结构，如下：

```
<!--[[flomo]]-->

{{#each}}

{{an_html}}

{{an_note}}

{{an_tags}}

{{/each}}
```

> 将上面的代码粘贴到下图：

![image](https://user-images.githubusercontent.com/81074/120091750-9e219200-c140-11eb-8a0d-faad20d6bfbd.png)

## 配置结束

> 截至到现在为止，全部的配置已经结束，接下来是如何使用。

## 使用

> 只要是支持导出的地方，均可使用，如：聚焦模式 / 阅读模式 / 稍后读（新布局）/ 反向链接等。

![image](https://user-images.githubusercontent.com/81074/120053980-df8e4080-c05f-11eb-8e5c-17fe750880c0.png)

## 备注

1. 当导入为全文时（没有标注时，`[[content]]` 的内容为空

## 引申

> 因为使用了 定制化 Markdown，可以将 **[永久链接](https://github.com/Kenshin/simpread/discussions/2221)** 也加入到 flomo 中，在这里就不展开描述了，后续我可以考虑再继续深入此教程。

> 因为使用了定制化 Markdown 功能，所以包含 **反向链接** 等内容通过定制化均可导入到 flomo，相比直接导入方式，更加灵活，但缺点是需要配置。

## 参考知识点

> 上述操作都已经写到了文档里面，有感兴趣的用户，可以看下。

- [定制化导出](http://ksria.com/simpread/docs/#/定制化导出)
- [Webhook](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=webhook)

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/29042336/123092196-cf019800-d45c-11eb-8da0-0e571f41e137.png)

**Comment by ghost** [comment]
诶嘿 我这边试了一下。
通过上述webhook代码 导出的 批注
是没有内容的.... content字段没有值。
flomo中只能看到网站title url 啥的...批注本身内容 莫得

**Reply by Kenshin** [reply]
正常是没问题的。造成这个情况的原因很多，但一般来说 **估计是你选择了导出全文**，看下图的设置

![image](https://user-images.githubusercontent.com/81074/123191858-7372f100-d4d4-11eb-8ba5-57b1606f328f.png)

**Reply by ghost** [reply]
emmmm 多谢解答，
不过我试了一下 还是没有 note content tags字段

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/29042336/123196626-d49ec280-d4dc-11eb-97e9-66e1ee66fcb2.png)

**Reply by ghost** [reply]
webhook 是直接复制 + 自己的flomoapi 的上文代码
![image](https://user-images.githubusercontent.com/29042336/123196838-34956900-d4dd-11eb-8dac-d4fd2cdda3b2.png)

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/29042336/123196947-5d1d6300-d4dd-11eb-93b0-30c02ff976c4.png)
然后 进入阅读模式 点的这个

**Reply by Kenshin** [reply]
emmm 很奇怪。在 Github 上找到我的邮箱，然后把配置文件发给我，我瞧瞧（最好含 flomo API）如觉得有安全问题，你可以随时重置 API

**Reply by Kenshin** [reply]
> ![image](https://user-images.githubusercontent.com/29042336/123196947-5d1d6300-d4dd-11eb-93b0-30c02ff976c4.png)
> 然后 进入阅读模式 点的这个

我好像知道了... 你需要先标注然后再导出，截图上似乎不是已被标注的内容。

**Reply by ghost** [reply]
这个也试过了 标记过也不是很行

**Reply by ghost** [reply]
邮件发咯~ 感谢~

**Reply by Kenshin** [reply]
好，确认后我会这里通知你。

**Reply by Kenshin** [reply]
嗨，知道是什么问题了，看下图

![image](https://user-images.githubusercontent.com/81074/123206361-23088d00-d4ee-11eb-9838-88b77e881117.png)

仔细看上文，你少加了一些内容

![image](https://user-images.githubusercontent.com/81074/123206418-3b78a780-d4ee-11eb-925d-50d54db9be7c.png)

**Reply by ghost** [reply]
嘿！成了！
还真是这个！
厉害了！我还以为这俩是分开的...哎还是不熟悉...呜呜呜呜呜呜 感谢大佬 这个应用这么爽，甚至还有售后也太神仙了

**Reply by Kenshin** [reply]
哈哈，没事，搞定就好。

神仙售后...  😅  我已经持续这种身份很多年了。

**Reply by ghost** [reply]
嘿嘿！之前第一次用简悦的时候 就给好评了。能和很多笔记联动这个真的是超实用。
另外很羡慕您的生活 能力强的独立开发者 感觉都很幸福同时很自由

**Reply by Kenshin** [reply]
哈哈，谢谢对简悦的支持。

做独立开发的一个好处是：可以通过自己的作品跟用户「交流」，甚至于很多都能成为朋友。

**Comment by winter60** [comment]
~~我测试了一下，上面的我都配置了，包括webhook 和定制化 Markdown 模板，但保存到flomo还是以前的格式。给您截个图可以帮忙看一下吗？~~(更正一下，webhook的按钮可以，是“保存到flomo”选项不行) 

这是我的配置：
![image](https://user-images.githubusercontent.com/23653326/167681491-1970c044-8d9b-4f6c-8e1c-a1384d1cacb7.png)

这是实际到flomo的效果：
![image](https://user-images.githubusercontent.com/23653326/167681568-b0436651-e298-4f82-8225-00036acb6458.png)

**Reply by Kenshin** [reply]
> 更正一下，webhook的按钮可以，是“保存到flomo”选项不行

是的，这个功能是针对 Webhook 的功能，无法使用 **导出到 flomo**，也就是说它们是两个功能。

**Reply by winter60** [reply]
> > 更正一下，webhook的按钮可以，是“保存到flomo”选项不行
> 
> 是的，这个功能是针对 Webhook 的功能，无法使用 **导出到 flomo**，也就是说它们是两个功能。

好的，谢谢作者大大回复，这点我明白了，但自己为什么markdown模板不起作用呢？
使用保存到flomo按钮和单条标注上的保存到flomo这两个用的都不是自己的模板，只有针对文档复制markdown到剪切板时`红色圈选处1`起了作用，如果让圈选处1和圈选处2起作用呢？
